src/games/strategy/net/ClientMessenger.java

```diff
@@ -111,7 +111,11 @@ public ClientMessenger(final String host, final int port, final String name, fin
       // our socket channel should already be closed
       m_socket.shutDown();
       if (conversation.getErrorMessage() != null) {
-        login.notifyFailedLogin(conversation.getErrorMessage());
+        String msg = conversation.getErrorMessage() ;
+        if(m_connectionRefusedError != null) {
+          msg += ", "+ m_connectionRefusedError;
+        }
+        login.notifyFailedLogin(msg);
```

Comment:
Really could use some better formatting here. Space after 'if', no space before ';', and space before '+'.